### ISSUE TITLE
InputTextMultiline causes crash with large maxLength values

### ISSUE THREAD

**Issue by SamVanheer** [mellinoe/ImGui.NET#102]
When calling `InputTextMultiline` with large `maxLength` values, such as `ushort.MaxValue`, the program will crash when the memory allocated with `AllocHGlobal` for `utf8InputBytes` is freed.

**Comment by mellinoe** [comment]
@SamVanheer I'm not repro'ing this, at least not precisely as written, e.g. this seems to work okay in the sample program:

```C#
private static string _testInput = "TestInputDefaultValue";
...
ImGui.InputTextMultiline("TestLabel", ref _testInput, ushort.MaxValue, new Vector2(200, 200))
```

Could you clarify in which scenario this causes a crash?

**Comment by SamVanheer** [comment]
My code is the same as yours, aside from the test input being a different string and the input area being smaller:

```
private static string _text = "test";
...
ImGui.InputTextMultiline("foo", ref _text, ushort.MaxValue, new Vector2(200, 20));
```

I am however running a 32 bit host. It's loading the Core Library from this path: `C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App\2.0.6\System.Private.CoreLib.dll`

So it could be 32 bit specific.

**Comment by mellinoe** [comment]
@SamVanheer Thanks for the hint. The crash does seem to only happen in 32-bit apps, but there was a buffer overflow happening regardless. This should be fixed in 1.66.1, with [this commit](https://github.com/mellinoe/ImGui.NET/commit/571f7975cdf02e448381290dd5d270cb576d3192).

**Comment by SamVanheer** [comment]
I can confirm that it works properly now.
